### ISSUE TITLE
URI path is not respected in Endpoint

### ISSUE THREAD

**Issue by j-chmielewski** [hyperium/tonic#1314]
### Version

tonic v0.8.3
tonic-build v0.8.4

### Platform

Linux blade 6.2.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 03 Mar 2023 15:58:31 +0000 x86_64 GNU/Linux

### Description

Tonic client does not respect endpoint path.

Trying to proxy my grpc service via nginx, I have the following nginx configuration:

```nginx
http {
    server {
        listen 50050 http2;
        server_name main;

	location /grpc {
	    grpc_pass grpc://core:50055;
        }
    }
}
```

And the client builds the endopoint like so:

```rust
let endpoint = Endpoint::from_static("http://localhost:50050/grpc");
```

I expected the request go to `/grpc` path but instead it does not - nginx logs show that a POST requests goes to `/` instead of `/grpc`. I also verified that changing nginx configuration to `location /` does work even though the client-side endpoint still indicates the `/grpc` path.

**Comment by LucioFranco** [comment]
Yeah, that seems like a bug here https://github.com/hyperium/tonic/blob/master/tonic/src/client/grpc.rs#L380 and maybe also on the server side. I would accept a PR that fixed this.

**Comment by democ98** [comment]
I also encountered the same problem, I don't know whether it is a server or client problem. Can someone explain it in detail? thanks

**Comment by mezeipetister** [comment]
So my understanding is this:

```rust
let channel = Channel::from_static("http://localhost").connect().await?;

  let mut client = InvoiceClient::with_origin(channel, Uri::from_static("/invoice_service"));
```

this way it works, but a but strange behavior.

**Comment by mezeipetister** [comment]
but this way its not possible to create an intercepted client with custom origin. Am I right?

**Comment by kumarUjjawal** [comment]
@LucioFranco has this been resolved, if not I would like to tackle it.

**Comment by LucioFranco** [comment]
@kumarUjjawal I believe this was fixed, feel free to see if you can reproduce this and either write a test or find a test (there may be one I forget).

**Comment by kumarUjjawal** [comment]
@LucioFranco Sure, Thank you!
